runtime/src/main/java/com/dylibso/chicory/runtime/Module.java

```diff
@@ -83,10 +83,17 @@ public Instance instantiate(HostImports hostImports) {
                     globals[i] = Value.f64(instr.getOperands()[0]);
                     break;
                 case GLOBAL_GET:
-                    // TODO this assumes that these are already initialized declared in order
-                    // should we make this more resilient? Should initialization happen later?
-                    globals[i] = globals[(int) instr.getOperands()[0]];
-                    break;
+                    {
+                        // TODO this assumes that these are already initialized declared in order
+                        // should we make this more resilient? Should initialization happen later?
+                        var globalImports = hostImports.getGlobals();
+                        var idx = (int) instr.getOperands()[0];
+                        globals[i] =
+                                idx < globalImports.length
```

Comment:
I like this implementation, so far I have been following a slightly different pattern as it was "more used" in the rest of the codebase: https://github.com/dylibso/chicory/blob/361c145ebf83b283003ed7053f3cdb09607f9661/runtime/src/main/java/com/dylibso/chicory/runtime/Machine.java#L303

just a nitpick.

Reply:
In this case though, instance is not yet available since it's constructed at the end of this method.

Reply:
Did you want to discuss this further, or are you happy with merging?

Reply:
good to go 👍  thanks!